statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -759,22 +759,52 @@ class VARProcess(object):
     Parameters
     ----------
     coefs : ndarray (p x k x k)
-    intercept : ndarray (length k)
+        coefficients for lags of endog, part or params reshaped
+    coefs_exog : ndarray
+        parameters for trend and user provided exog
     sigma_u : ndarray (k x k)
+        residual covariance
     names : sequence (length k)
+    _params_info : dict
+        internal dict to provide information about the composition of `params`,
+        specifically `k_trend` (trend order) and `k_exog_user` (the number of
+        exog variables provided by the user).
+        If it is None, then coefs_exog are assumed to be for the intercept and
+        trend.
 
     Returns
     -------
     **Attributes**:
     """
-    def __init__(self, coefs, exog, sigma_u, names=None):
+    def __init__(self, coefs, coefs_exog, sigma_u, names=None, _params_info=None):
         self.k_ar = len(coefs)
         self.neqs = coefs.shape[1]
         self.coefs = coefs
-        self.exog = exog
+        self.coefs_exog = coefs_exog
+        # Note reshaping 1-D coefs_exog to 2_D makes unit tests fail
         self.sigma_u = sigma_u
         self.names = names
 
+        if _params_info is None:
+            _params_info = {}
+        self.k_exog_user = _params_info.get('k_exog_user', 0)
+        if self.coefs_exog is not None:
+            k_ex = self.coefs_exog.shape[0] if self.coefs_exog.ndim != 1 else 1
+            k_c = k_ex - self.k_exog_user
+        else:
+            k_c = 0
+        self.k_trend = _params_info.get('k_trend', k_c)
```

Comment:
It looks like "k_trend" and "k_exog_user" are the only keys that are relevant here.  Why not just make those optional kwargs to `__init__`?  It's clearer and doesn't risk silently ignoring anything.

Reply:
because adding `k_trend` explicitly is not **future proof**. We would have to change the signature again if or when we add seasonal, k_seasonal or other deterministic or exog with explicit built in support.

Reply:
Well that's... pretty silly.  Nothing is future proof, and it's OK to change the signature when behavior changes.  Better to do that than to de-facto change the signature in an opaque and error-prone way.

But you're going to reflexively argue with any suggestion I give, and this isn't worth the time.  Do whatever you want.